src/makeInfinityAdder.js

```diff
@@ -4,7 +4,26 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+  let result = 0;
+
+  function adder(num) {
+    if (num === undefined) {
+      result = sum;
+
+      sum = 0;
+
+      return result;
+    } else {
+      sum += num;
+
+      result = sum;
```

Comment:
The `result` variable is not needed here. You can remove this line to simplify the code.